### PR TITLE
deprecate co recipe in favor of cocoyaxi

### PR DESCRIPTION
**Pull request by SpaceIm**
see https://github.com/conan-io/conan-center-index/pull/8915#discussion_r784811935

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/co/all/conanfile.py [hunk 1]
```diff
@@ -12,6 +11,9 @@ class CoConan(ConanFile):
     license = "MIT"
     description = "A go-style coroutine library in C++11 and more."
     topics = ("co", "coroutine", "c++11")
+
+    deprecated = "cocoyaxi"
```
Comment (SpaceIm) [inline recipes/co/all/conanfile.py:15]: I'm wondering: should there be `provides` also?

Comment (idealvin) [reply recipes/co/all/conanfile.py:15]: I'm not familar with conan. So what I should do now?

Comment (idealvin) [reply recipes/co/all/conanfile.py:15]: > I'm wondering: shouldn't there be `provides` also?

I have added cocoyaxi in #8930

Comment (idealvin) [reply recipes/co/all/conanfile.py:15]: For co 2.0.2, the sha256 sum should be 5dbb90e18627ebe563a6a437e8e28329a6bfc07f266718318bd3adf26027d156

**Review by idealvin — DISMISSED** [review]

**Comment by SSE4** [comment]
was it rebranded upstream, or what exactly happened? I can't get from #8915

**Comment by SpaceIm** [comment]
see https://github.com/conan-io/conan-center-index/pull/8930

**Comment by prince-chrismc** [comment]
It was renamed, https://github.com/idealvin/co points to new name

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Comment by SSE4** [comment]
> It was renamed, https://github.com/idealvin/co points to new name

thanks for the clarification!

**Review by uilianries — APPROVED** [review]

**Pull request merged**